syntax = "proto3";

message VaultRequest {
    string vault_token = 1;
    string requested_secret = 2;
}

message KeyValuePair {
    string key = 1;
    string value = 2;
}

message VaultResponse {
    repeated KeyValuePair secrets = 1;
}

service VaultManager {
    rpc get_secret(VaultRequest) returns (VaultResponse);
}